src/components/v5/frame/ColonyHome/partials/DashboardHeader/partials/ColonyLinks/useHeaderLinks.ts

```diff
@@ -0,0 +1,145 @@
+import { CopySimple, Door, Rocket, ShareNetwork, Smiley } from 'phosphor-react';
+import { useLocation } from 'react-router-dom';
+
+import { useColonyDashboardContext } from '~context/ColonyDashboardContext';
+import useColonyContext from '~hooks/useColonyContext';
+import { useCopyToClipboard } from '~hooks/useCopyToClipboard';
+import useMobile from '~hooks/useMobile';
+import { COLONY_DETAILS_ROUTE } from '~routes';
+import { formatText } from '~utils/intl';
+import {
+  DropdownMenuProps,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+} from '~v5/common/DropdownMenu/types';
+import { COLONY_LINK_CONFIG } from '~v5/shared/SocialLinks/colonyLinks';
+
+import { sortExternalLinks } from './helpers';
+
+export const useHeaderLinks = (): { dropdownMenuProps: DropdownMenuProps } => {
+  const isMobile = useMobile();
+
+  const { colony, colonySubscription } = useColonyContext();
+  const { pathname } = useLocation();
+  const { openLeaveColonyModal } = useColonyDashboardContext();
+
+  const {
+    handleClipboardCopy: handleShareUrlItemCopy,
+    isCopied: isShareUrlItemCopied,
+  } = useCopyToClipboard(5000);
+  const {
+    handleClipboardCopy: handleColonyAddressItemCopy,
+    isCopied: isColonyAddressItemCopied,
+  } = useCopyToClipboard(5000);
+
+  const colonyUrl = `${window.location.host}${pathname}`;
+  const { isWatching } = colonySubscription;
+  const { colonyAddress, metadata } = colony || {};
+
+  const externalLinks = metadata?.externalLinks
+    ? sortExternalLinks(metadata.externalLinks)
+    : [];
+
+  const menuItems: DropdownMenuGroup[] = [
+    {
+      key: '1',
```

Comment:
You mentioned there was a PR where we would tackle all of these "keys", right ?

Reply:
nice catch, will do them now after I've also rebased this PR on master!